src/common/transformations/src/transformations/common_optimizations/reduce_reshape_fusion.cpp

```diff
@@ -0,0 +1,78 @@
+    MATCHER_SCOPE(ReduceReshapeFusion);
+
+    const auto reduce_axes = pattern::wrap_type<opset9::Constant>();
+    const auto reduce = pattern::wrap_type<op::util::ArithmeticReductionKeepDims, op::util::LogicalReductionKeepDims>(
+        {pattern::any_input(), reduce_axes},
+        pattern::has_static_shape());
+    const auto reshape =
+        pattern::wrap_type<opset9::Reshape>({reduce, pattern::any_input()}, pattern::has_static_shape());
+
+    matcher_pass_callback callback = [=](pattern::Matcher& m) {
+        auto& pattern_map = m.get_pattern_value_map();
+        auto reshape_node = pattern_map.at(reshape).get_node_shared_ptr();
+        const auto reduce_node = pattern_map.at(reduce).get_node_shared_ptr();
+        const bool keep_dims =
+            std::dynamic_pointer_cast<op::util::ArithmeticReductionKeepDims>(reduce_node)
+                ? std::dynamic_pointer_cast<op::util::ArithmeticReductionKeepDims>(reduce_node)->get_keep_dims()
+                : std::dynamic_pointer_cast<op::util::LogicalReductionKeepDims>(reduce_node)->get_keep_dims();
+
+        if (!reduce_node) {
+            return false;
+        }
+
+        if (keep_dims) {
+            return false;
+        }
+
+        const auto reduce_axes_val = std::dynamic_pointer_cast<op::util::ArithmeticReductionKeepDims>(reduce_node)
+                                         ? std::dynamic_pointer_cast<op::util::ArithmeticReductionKeepDims>(reduce_node)
+                                               ->get_reduction_axes()
+                                               .to_vector()
+                                         : std::dynamic_pointer_cast<op::util::LogicalReductionKeepDims>(reduce_node)
+                                               ->get_reduction_axes()
+                                               .to_vector();
+        const auto& reshape_shape = reshape_node->get_shape();
+
+        auto reduce_shape_if_keep_dims = reduce_node->get_shape();
+        for (const auto& axis : reduce_axes_val) {
+            reduce_shape_if_keep_dims.insert(std::next(std::begin(reduce_shape_if_keep_dims), axis), 1);
+        }
+
+        if (reduce_shape_if_keep_dims != reshape_shape) {
+            return false;
+        }
+
+        if (auto arithmetic_reduce_node =
+                std::dynamic_pointer_cast<op::util::ArithmeticReductionKeepDims>(reduce_node)) {
+            arithmetic_reduce_node->set_keep_dims(true);
+        } else if (auto logical_reduce_node =
+                       std::dynamic_pointer_cast<op::util::LogicalReductionKeepDims>(reduce_node)) {
+            logical_reduce_node->set_keep_dims(true);
+        }
+        reduce_node->set_friendly_name(reshape_node->get_friendly_name());
+        copy_runtime_info(reshape_node, reduce_node);
+        replace_node(m.get_match_root(), reduce_node);
+
+        return true;
+    };
+
+    auto m = std::make_shared<pattern::Matcher>(reshape, matcher_name);
+    this->register_matcher(m, callback);
```

Comment:
Is `this` needed here? (same question to the other file)

Reply:
removed